modules/indexing/src/main/java/org/apache/ignite/internal/visor/verify/ValidateIndexesClosure.java

```diff
@@ -206,53 +230,55 @@ private VisorValidateIndexesJobResult call0() {
 
         List<Future<Map<PartitionKey, ValidateIndexesPartitionResult>>> procPartFutures = new ArrayList<>();
         List<Future<Map<String, ValidateIndexesPartitionResult>>> procIdxFutures = new ArrayList<>();
+        List<T3<CacheGroupContext, GridDhtLocalPartition, Future<CacheSize>>> cacheSizeFutures = new ArrayList<>();
+        List<T3<GridCacheContext, Index, Future<T2<Throwable, Long>>>> idxSizeFutures = new ArrayList<>();
+
         List<T2<CacheGroupContext, GridDhtLocalPartition>> partArgs = new ArrayList<>();
         List<T2<GridCacheContext, Index>> idxArgs = new ArrayList<>();
 
         totalCacheGrps = grpIds.size();
 
         Map<Integer, IndexIntegrityCheckIssue> integrityCheckResults = integrityCheckIndexesPartitions(grpIds);
 
+        GridQueryProcessor qryProcessor = ignite.context().query();
+        IgniteH2Indexing h2Indexing = (IgniteH2Indexing)qryProcessor.getIndexing();
+
         for (Integer grpId : grpIds) {
             CacheGroupContext grpCtx = ignite.context().cache().cacheGroup(grpId);
 
-            if (grpCtx == null || integrityCheckResults.containsKey(grpId))
+            if (isNull(grpCtx) || integrityCheckResults.containsKey(grpId))
```

Comment:
I think that means '--check-size' will not work for in memory regions and without CRC checking.
Did this covered tests?

Reply:
We get map integrityCheckResults from
org.apache.ignite.internal.visor.verify.ValidateIndexesClosure#integrityCheckIndexesPartitions
and we don 't filter out caches with memory regions and CRC checking perform for cache with persistent regions.

Reply:
Added tests for a non- persistent region.
org.apache.ignite.util.GridCommandHandlerIndexingCheckSizeTest#testCheckCacheSizeWhenBrokenCacheInNonPersistRegion
org.apache.ignite.util.GridCommandHandlerIndexingCheckSizeTest#testCheckCacheSizeWhenBrokenIdxInNonPersistRegion